src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -432,6 +433,17 @@ private void InitializeOutputFile(TOptions analyzeOptions, TContext context, ISe
 
                         SarifLogger sarifLogger;
 
+                        _run = new Run();
+
+                        if (!string.IsNullOrWhiteSpace(analyzeOptions.AutomationId) || !string.IsNullOrWhiteSpace(analyzeOptions.AutomationGuid))
+                        {
```

Comment:
Do you actually need this check? i.e., can't we just create the object and if both items are null it doesn't get emitted?

If that's not true, change the RunAutomationDetails ShouldSerializeXXX methods to make that happen. Add tests for this, too, if they don't exist.

Reply:
I had to add three ShouldSerialize:
1. ShouldSerializeAutomationDetails
2. ShouldSerializeId
3. ShouldSerializeGuid

if id and guid are null/empty, we should not emit automation details and that check can only occur in the 'run'.

Once we have an AutomationDetails, if one of those is empty, that property should not be emitted.